### PR TITLE
ARSSTB-285 [tech debt]: remove dupe files, tidy up

### PR DESCRIPTION
**Pull request by C-Thomp**
- tests that failed with agents flag fixed
- removed unused pages: WhoAreYouAgent
- userrole navigation around CheckEORIDetails now tested
- Html View cleanup and some messages updates
- html in messages file removed
-  not longer handling the check TRADER eoridetails navigation as this will be covered elsewhere (didn't want to bloat this too much)

### Type of change
- [x] Non-breaking change

### Description
Some tech debt and clean up spotted

### CODE REVIEW

File: app/views/AgentForTraderCheckRegisteredDetailsView.scala.html [hunk 1]
```diff
@@ -134,10 +132,10 @@
     ).withHint(Hint(content = HtmlContent(paraContent)))
 )
 
-@searchLink
+    @paragraph(content = Html(messages("traderDetails.common.searchAgain", searchAgainLink)))
```
Comment (aaocs) [inline app/views/AgentForTraderCheckRegisteredDetailsView.scala.html:135]: Is the call to searchAgainLink able to find traderDetails.common.searchAgain.link automatically?

File: test/base/SpecBase.scala [hunk 2]
```diff
@@ -70,11 +70,17 @@ trait SpecBase
...
+
+  val userAnswersAsOrgAdmin: UserAnswers =
+    emptyUserAnswers
+      .setFuture(AccountHomePage, OrganisationAdmin)
+      .futureValue
 
-  val userAnswersAsOrgAdmin: UserAnswers = emptyUserAnswers
+  emptyUserAnswers
     .setFuture(AccountHomePage, OrganisationAdmin)
     .futureValue
```
Comment (aaocs) [inline test/base/SpecBase.scala:85]: This looks like duplication of userAnswersAsOrgAdmin. Maybe set emptyUserAnswers to private to force the other user answers to be used?

Comment (C-Thomp) [reply test/base/SpecBase.scala]: oops, yep a dupe

File: app/views/AgentForTraderCheckRegisteredDetailsView.scala.html (hunk 1 above)
Comment (aaocs) [reply app/views/AgentForTraderCheckRegisteredDetailsView.scala.html:135]: Ah ok, thanks for the explanation.

**Review by aaocs — DISMISSED** [review]
This all looks great :D 
Verified the journey works up to "Upload your letter of authority" screen

File: app/forms/WhoAreYouAgentFormProvider.scala
Comment (paulhargrave) [inline app/forms/WhoAreYouAgentFormProvider.scala:1]: Nice - I'm very glad this has finally disappeared!

File: app/navigation/Navigator.scala [hunk 3]
```diff
@@ -85,14 +85,14 @@ class Navigator @Inject() (appConfig: FrontendAppConfig, userRoleProvider: UserR
     case ExplainHowYouWillUseMethodSixPage                => explainHowYouWillUseMethodSixPage
     case AdaptMethodPage                                  => adaptMethodPage
     case DeleteDraftPage                                  => _ => AccountHomeController.onPageLoad()
-    case WhoAreYouAgentPage                               => whoAreYouRouting
```
Comment (paulhargrave) [inline app/navigation/Navigator.scala:88]: woohoo

**Review by paulhargrave — APPROVED** [review]
Excellent work. Yeah as you said this maybe could have separate tickets. These things always seem to grow.... 

Thanks Carl

**Review by janosgyenge-digital — APPROVED** [review]

**Pull request merged**